data/data/azure/main.tf

```diff
@@ -35,21 +37,32 @@ module "bootstrap" {
 
   # This is to create explicit dependency on private zone to exist before VMs are created in the vnet. https://github.com/MicrosoftDocs/azure-docs/issues/13728
   private_dns_zone_id = azurerm_dns_zone.private.id
+
+  preexisting_network    = var.azure_preexisting_network
 }
 
 module "vnet" {
   source              = "./vnet"
-  vnet_name           = azurerm_virtual_network.cluster_vnet.name
+  vnet_name           = local.virtual_network
   resource_group_name = azurerm_resource_group.main.name
   vnet_cidr           = var.machine_cidr
   master_subnet_cidr  = local.master_subnet_cidr
-  node_subnet_cidr    = local.node_subnet_cidr
+  worker_subnet_cidr  = local.worker_subnet_cidr
   cluster_id          = var.cluster_id
   region              = var.azure_region
   dns_label           = var.cluster_id
 
   # This is to create explicit dependency on private zone to exist before VMs are created in the vnet. https://github.com/MicrosoftDocs/azure-docs/issues/13728
   private_dns_zone_id = azurerm_dns_zone.private.id
+
+  preexisting_network    = var.azure_preexisting_network
+  network_resource_group = var.azure_network_resource_group_name
+  virtual_network        = local.virtual_network
+  master_subnet          = var.azure_control_plane_subnet
+  worker_subnet          = var.azure_compute_subnet
+
+  master_subnet_id       = module.vnet.master_subnet_id
```

Comment:
why are we passing output of this module as input to the module???

Reply:
Clearly I didn't know how modules work. I've fixed this ;-)